product-web-app/middleware/auth.ts

```diff
@@ -0,0 +1,22 @@
+import { Application, NextFunction, Request, Response } from 'express';
+import axios from 'axios';
+
+export default class AuthMiddleware {
+  appRoutes(app: Application) {
+    app.use((req: Request, res: Response, next: NextFunction) => {
+      if (req.path === '/auth/register') {
+        next();
+      } else if (req.cookies.token) {
+        axios.defaults.headers.common.Authorization = req.cookies.token;
+        req.session.isAdmin = req.cookies.roleId;
```

Comment:
Can you try to use
app.locals
http://expressjs.com/en/api.html
They should be automatically added to the render function and available in the nunjuck templates